django/utils/html.py

```diff
@@ -324,6 +325,7 @@ def __call__(self, text, trim_url_limit=None, nofollow=False, autoescape=False):
             ]
         )
 
+    @functools.lru_cache(maxsize=1000)
```

Comment:
Thought: `lru_cache` with a `maxsize` has an internal thread lock, could this impact performance? Given `handle_word` could be called multiple times per `urlize`, which in turn could be called many more times per request, the lock could get quite contended in a multi-threaded environment. Is it worth checking benchmarks?